frontend/src/views/student/discussions/ReplyComponent.vue

```diff
@@ -0,0 +1,205 @@
+              <v-expansion-panel-content>
+                <div
+                  v-for="reply in discussion.replies"
+                  :key="reply.id"
+                  class="text-left reply"
+                >
+                  <div style="width: 100%">
+                    <div>
+                      <b v-if="user.id !== reply.userId"
+                        >{{ reply.userName }} replied on {{ reply.date }} :
+                      </b>
+                      <b v-else>You replied on {{ reply.date }} :</b>
+                      <span v-html="convertMarkDown(reply.message)" />
+                    </div>
+                  </div>
+                </div>
+                <div
+                  class="reply-message"
+                  v-if="!discussion.closed && discussion.userId === user.id"
+                >
+                  <v-textarea
+                    data-cy="replyTextArea"
+                    class="textarea-reply"
+                    solo
+                    :id="'reply' + discussion.userId"
+                    label="Type a reply..."
+                    @input="setReplyMessage"
+                  ></v-textarea>
+                  <v-card-actions>
+                    <v-btn
+                      data-cy="submitReplyButton"
+                      class="submit-button"
+                      @click="
+                        submitReply();
+                        clearTextarea('#reply' + discussion.userId);
+                      "
+                      >Submit</v-btn
+                    >
+                  </v-card-actions>
+                </div>
+              </v-expansion-panel-content>
+            </v-expansion-panel>
+          </v-expansion-panels>
+          <div v-else class="reply-message">
+            <v-textarea
+              class="textarea-reply"
+              solo
+              :id="'reply' + discussion.userId"
+              label="Type a reply..."
+              @input="setReplyMessage"
+            ></v-textarea>
+            <v-card-actions>
+              <v-btn
+                class="submit-button"
+                @click="
+                  submitReply();
+                  clearTextarea('#reply' + discussion.userId);
+                "
+                >Submit</v-btn
+              >
+            </v-card-actions>
+          </div>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Emit } from 'vue-property-decorator';
+import { convertMarkDown } from '@/services/ConvertMarkdownService';
+import Discussion from '@/models/management/Discussion';
+import RemoteServices from '../../../services/RemoteServices';
+import User from '@/models/user/User';
+
+@Component
+export default class ReplyComponent extends Vue {
+  @Prop() readonly discussion!: Discussion;
+  replyMessages: Map<number, string> = new Map();
+  user: User = this.$store.getters.getUser;
+
+  @Emit()
+  async submitReply() {
```

Comment:
This function is way more complex than it needs to be, due to a change in a feature that I initially implemented, namely: Students can answer to other students discussions. Please don't pay attention to it, I will change after the frontend feedback.